drivers/serial/pty.c

```diff
@@ -481,96 +492,18 @@ static ssize_t pty_read(FAR struct file *filep, FAR char *buffer, size_t len)
        * translations.
        */
 
-      ntotal = 0;
-      for (i = 0; i < len; i++)
+again:
```

Comment:
Optional: I think it is better to place `again:` few lines above before `ntotal = file_read(&dev->pd_src, buffer, len);`.